bokehjs/src/lib/models/tools/edit/point_draw_tool.ts

```diff
@@ -64,8 +64,9 @@ export class PointDrawToolView extends EditToolView {
     if (!this.model.drag) { return; }
     this._pan(ev);
     for (const renderer of this.model.renderers) {
-      renderer.data_source.selected.indices = [];
-      renderer.data_source.properties.data.change.emit();
+      renderer.data_source.selected.indices = []
+      renderer.data_source.data = renderer.data_source.data
+      renderer.data_source.properties.data.change.emit()
     }
```

Comment:
self assign required to notify documemt (to sync with python)

Reply:
Thanks, I had previously assumed ``renderer.data_source.properties.data.change.emit()`` was sufficient.

Reply:
it's definitely a weird edge, should probably be documented in the dev guide somewhere